Source/WebKit/WPEPlatform/wpe/drm/WPEDRMCursor.cpp

```diff
@@ -69,11 +69,11 @@ void Cursor::updateBuffer(const uint8_t* pixels, uint32_t width, uint32_t height
     RELEASE_ASSERT(width <= m_deviceWidth);
     RELEASE_ASSERT(height <= m_deviceHeight);
 
-    uint32_t deviceBuffer[m_deviceWidth * m_deviceHeight];
-    memset(deviceBuffer, 0, sizeof(deviceBuffer));
+    Vector<uint32_t> deviceBuffer;
+    deviceBuffer.reserveCapacity(m_deviceWidth * m_deviceHeight);
     for (uint32_t i = 0; i < height; ++i)
-        memcpy(deviceBuffer + i * m_deviceWidth, pixels + i * stride, stride);
-    gbm_bo_write(m_buffer->bufferObject(), deviceBuffer, sizeof(deviceBuffer));
+        memcpy(&deviceBuffer[i * m_deviceWidth], pixels + i * stride, stride);
+    gbm_bo_write(m_buffer->bufferObject(), deviceBuffer.data(), deviceBuffer.sizeInBytes());
```

Comment:
Hm, does this really work? The capacity of the vector is not the same as its size. No elements are added to the vector, so even if setting the capacity is sufficient to avoid memory corruption, I assume its size is still zero at the end here? I think you want to instead set the initial size. `Vector<uint32_t> deviceBuffer(m_deviceWidth * m_deviceHeight);` should get you a vector of the correct size with uninitialized values.

Reply:
Ah yes I think you're right.